torch/_subclasses/fake_impls.py

```diff
@@ -455,9 +457,28 @@ def masked_select(fake_mode, func, self, mask):
         has_free_symbols,
     )
 
+    # If num elements is expressed symbolically, calculate
+    # the concrete value based on upper bounds. Otherwise,
+    # we can set max val directly.
     if not has_free_symbols(self.numel()):
-        if self.numel() > 2:
-            maxval = int(self.numel())
+        num_elements = int(self.numel())
+    else:
+        if len(self.shape) != 0:
+            num_elements = 1
+            for size in self.shape:
+                if isinstance(size, int):
+                    num_elements *= size
+                elif isinstance(size, torch.SymInt):
+                    sym_node = size.node
+                    sym_range = bound_sympy(
+                        sym_node.expr, sym_node.shape_env.var_to_range
+                    )
+                    if isinstance(sym_range.upper, IntInfinity):
+                        num_elements = sys.maxsize - 1
```

Comment:
Why don't you just use int infinity here?

Reply:
Keeping consistent with https://github.com/pytorch/pytorch/blob/ea89f012817d54a1b858d8f62f0f3deea27d008a/torch/_subclasses/fake_impls.py#L450

Reply:
Is out of date but no matter, can be updated separately